static.js

```diff
@@ -31,18 +35,14 @@ if (allowSave)
 
 http.createServer(function(req, res) {
     var uri = unescape(url.parse(req.url).pathname);
+    console.log("[" + req.method + "]", uri);
+    var root = process.cwd() + path.sep;
+    var filename = path.join(root, uri);
 
-    // We don't allow for relative URIs, such as ../../X, to prevent directory traversal in case the server is shared with other actors.
-    // In windows, `path.normalize` converts `/` in the URI to `\\`, so we enforce POSIX path separators during the check.
-    // See more at https://cwe.mitre.org/data/definitions/22.html
-    var posixUri = uri.replaceAll(path.sep, path.posix.sep);
-    var normalizedPosixUri = path.normalize(uri).replaceAll(path.sep, path.posix.sep);
-    if (normalizedPosixUri !== posixUri) {
+    if (filename.slice(0, root.length) !== root) {
```

Comment:
Won't a path like "path/to/ace/folder/../../../../CONFIDENTIAL_FILE" pass this check?

Reply:
What if we normalize it and then check if it doesn't start with root?

Reply:
Path join already calls normalize, `The path.join() method joins all given path segments together using the platform-specific separator as a delimiter, then normalizes the resulting path.` https://nodejs.org/api/path.html#pathjoinpaths